### ISSUE TITLE
Add date i18n to Grav Core

### ISSUE THREAD

**Issue by tdechiron** [getgrav/grav#2926]
Hello there,

I am a french developer who is trying Grav CMS to replace Wordpress projects.

From now I can do whatever I want and I am pretty happy with Grav, there is just one point on **date internationalization**.
I can assure you this is a very important problem for foreign users.

From the Grav manual, there is an explanation on the problem and fixes for it : 
https://learn.getgrav.org/16/cookbook/twig-recipes#displaying-a-translated-month

From my point of view this fixes are not correct : 
- you should not have to install a plugin for i18n of dates
- the 2 solutions are not using server locales nor date formats from configuration
- in configuration **override_locale** is useless

An another problem is date format i18n : you could have different date format depending on localization.

### 1. use strftime for date formatting instead of DateTime format : 

Install and set **teraone/twig-strftime-extension** in Grav
By default strftime format is localized, so replace Twig date filter by strftime filter do the job.
strftime format is very different of date format so this is a problem on backward compatibility.

### 2. use PHP Intl in core : 

Like the plugin https://github.com/Perlkonig/grav-plugin-twig-extensions
Integrate Intl to Grav and create a Twig filter like localizeddate() based on datetime object (like actual Twig date) and without the need to specify the locale (it should take the default locale or the current locale if **override_locale** is set).

## Date format i18n

This is a less critical problem, it can be solve by using languages.yaml in your theme, specifying date formats for each languages, and get date format in Twig : 
`t('MY_THEM.DATEFORMAT.LONG')`

A better approach should be to create a Twig function to return date format for any language (if **override_locale** is set in configuration) or the default date formats.

The current list of date formats in Admin plugin could be completed.

**Comment by ricardo118** [comment]
i agree with these points, having to work on many multilang websites with grav. 

In the meantime, i am using the localizeddate extension too to make things work like this:

`{{ "now"|localizeddate('long', 'none', grav.language.getActive()) }}`

**Comment by tdechiron** [comment]
Thanks for your message, after looking at **Intl functions**, I think using it is the correct solution, because it can handle datetime formats too (so adding it will allow backward compatibility with configuration date formats).
IntlDateFormatter formats are i18n, so it solve date formats i18n too.

I am thinking of a rewrite of **localizeddate filter** with less parameters : 
`localizeddate( $format, $pattern=null )`

**$format** could be : 
- 'config_default', 'config_short', 'config_long' (date formats of configuration)
- 'none', 'short', 'medium', 'long', 'full' (IntlDateFormatter formats)
- null (for pattern specific)

set **$pattern** like 'mm/dd/yyy'

the locale would be automatically set depending on **override_locale**

It could be interesting to rewrite **localizednumber** and **localizedcurrency** from the same plugin and add them to Grav Core too.

**Comment by ricardo118** [comment]
yea if we are to integrate something like that into core we would make the parameters a bit cleaner imo

**Comment by ricardo118** [comment]
ok so i spoke to andy for the go ahead and i will begin to integrate it into Grav core

**Comment by OleVik** [comment]
It would be advantageous to do this with a thought to fallbacks, ie. only load a third-party library if the Intl-extension is missing, and further fall back to the generic date-parser if the third-party library is not installed.

Two especially well-developed and -maintained libraries for this are [Carbon](https://carbon.nesbot.com/docs/#api-localization) and [Date](https://github.com/jenssegers/date#usage), where the latter extends the format and possibilities of the former. It would also be beneficial to focus on `localizeddate`, as suggested, but also create some thin wrappers in Twig for similar functionality - like localized formats for time ago, parse by locality, time differences, various getters, etc.
